onedal/spmd/linear_model/incremental_linear_model.py

```diff
@@ -0,0 +1,97 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+
+from daal4py.sklearn._utils import get_dtype
+
+from ...common.hyperparameters import get_hyperparameters
+from ...datatypes import _convert_to_supported, to_table
+from ...linear_model import (
+    IncrementalLinearRegression as base_IncrementalLinearRegression,
+)
+from ...utils import _check_X_y, _num_features
+from .._base import BaseEstimatorSPMD
+
+
+class IncrementalLinearRegression(BaseEstimatorSPMD, base_IncrementalLinearRegression):
+    """
+    Distributed incremental Linear Regression oneDAL implementation.
+
+    API is the same as for `onedal.linear_model.IncrementalLinearRegression`.
+    """
+
+    def _reset(self):
+        self._partial_result = super(base_IncrementalLinearRegression, self)._get_backend(
+            "linear_model", "regression", "partial_train_result"
+        )
+
+    def partial_fit(self, X, y, queue=None):
```

Comment:
I am still a little confused why this is re-implemented and cannot take the base estimator's - can you clarify?

Reply:
we don't have spmd partial_fit on c++ side, that's why it is reimplemented here to take non-spmd backend